lab/experiments/migrations/0036_alter_experiment_defaultcriteria.py

```diff
@@ -0,0 +1,39 @@
+# Generated by Django 4.0.7 on 2022-11-30 14:34
+
+from django.db import migrations, models
+import django.db.models.deletion
+
+
+def select_default_criterion(apps, schema_editor):
+    """If there are experiments in the db without defaultcriteria give it a
+    default DefaultCriteria
+    """
+    Experiment = apps.get_model("experiments", "Experiment")
+    DefaultCriteria = apps.get_model("experiments", "DefaultCriteria")
+    db_alias = schema_editor.connection.alias
+    experiments = Experiment.objects.using(db_alias).all()
+
+    for exp in experiments:
+        if exp.defaultcriteria is None:
+            def_id = DefaultCriteria.objects.using(db_alias).create().id
+            exp.defaultcriteria_id = def_id
+            exp.save()
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('experiments', '0035_remove_defaultcriteria_experiment_and_more'),
+    ]
+
+    operations = [
+        migrations.RunPython(select_default_criterion, None),
```

Comment:
You've made the effort of making 0035 reversible, but then 0036 isn't. Not too important as the db can be recreated from scratch for now

Reply:
Hmm apparently 0035 is also not really reversible as it fails on DefaultCriteria.experiment requiring a non-null value (when reversing RemoveField).
To get it to run you need to go to 0006 and add `null=True` to the experiment field. Thanks Django.